code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -32,6 +32,13 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food)
 				return 1
 		return 0
 
+	proc/get_food_color()
+		if (food_color) // keep manually defined food colors
+			return food_color
+		var/icon/I = istype(src.icon, /icon) ? src.icon : icon(src.icon, src.icon_state)
+		food_color = get_average_color(I)
+		return food_color
```

Comment:
Hmm, this caches the color which is probably good as it results in skipping the get_average_color call in future calls on this objects. However, note that this also means that food_color won't get updated if this changes icon_state. Though that's probably unlikely to matter much.

I'd keep it like you wrote it, this comment is more of a sidenote.

Reply:
yeah, ultimately some of the food procs check the color several times and it seemed more apt to just cache it than rebuild it.